scripts/update_lock.py

```diff
@@ -0,0 +1,206 @@
+#!/usr/bin/env python3
+
+# Copyright 2021 - 2023 Universität Tübingen, DKFZ, EMBL, and Universität zu Köln
+# for the German Human Genome-Phenome Archive (GHGA)
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+"""Update the dependency lock files located at 'requirements.txt' and
+'requirements-dev.txt'.
+"""
+
+import os
+import re
+import subprocess
+from copy import deepcopy
+from pathlib import Path
+from tempfile import TemporaryDirectory
+
+import tomli
+import tomli_w
+from script_utils import cli
+
+REPO_ROOT_DIR = Path(__file__).parent.parent.resolve()
+
+PYPROJECT_TOML_PATH = REPO_ROOT_DIR / "pyproject.toml"
+DEV_DEPS_PATH = REPO_ROOT_DIR / "requirements-dev.in"
+PROD_DEPS_PATH = REPO_ROOT_DIR / "requirements.in"
+OUTPUT_LOCK_PATH = REPO_ROOT_DIR / "requirements.txt"
+OUTPUT_DEV_LOCK_PATH = REPO_ROOT_DIR / "requirements-dev.txt"
+
+
+def exclude_from_dependency_list(*, package_name: str, dependencies: list) -> list:
+    """Exclude the specified package from the provided dependency list."""
+
+    return [
+        dependency
+        for dependency in dependencies
+        if not dependency.startswith(package_name)
+    ]
+
+
+def remove_self_dependencies(pyproject: dict) -> dict:
+    """Filter out self dependencies (dependencies of the package on it self) from the
+    dependencies and optional-dependencies in the provided pyproject metadata."""
+
+    if not "project" in pyproject:
```

Comment:
```suggestion
    if "project" not in pyproject:
```

a bit more idiomatic